### PR TITLE
Python Task Library error handling update for remaining tasks

### PR DESCRIPTION
**Pull request by kinaryml**
Incorporated the new error handling changes to the following tasks
- AudioEmbedder
- ImageClassifier
- ObjectDetector

### CODE REVIEW

File: tensorflow_lite_support/python/task/audio/pybinds/_pywrap_audio_embedder.cc [hunk 1]
```diff
@@ -14,9 +14,9 @@ limitations under the License.
 ==============================================================================*/
 
 #include "pybind11/pybind11.h"
-#include "pybind11_abseil/status_casters.h"  // from @pybind11_abseil
 #include "pybind11_protobuf/native_proto_caster.h"  // from @pybind11_protobuf
 #include "tensorflow_lite_support/cc/port/statusor.h"
```
Comment (khanhlvg) [inline tensorflow_lite_support/python/task/audio/pybinds/_pywrap_audio_embedder.cc:18]: This is no longer necessary.

File: tensorflow_lite_support/python/task/audio/pybinds/BUILD [hunk 2]
```diff
@@ -15,12 +15,12 @@ pybind_extension(
     module_name = "_pywrap_audio_embedder",
     deps = [
         "//tensorflow_lite_support/cc/port:statusor",
```
Comment (khanhlvg) [inline tensorflow_lite_support/python/task/audio/pybinds/BUILD:17]: This is no longer necessary.

**Review by khanhlvg — CHANGES_REQUESTED** [review]
Could you also update the _pywrap_audio_buffer.cc?

File: tensorflow_lite_support/python/task/audio/pybinds/_pywrap_audio_embedder.cc (hunk 1 above)
Comment (kinaryml) [reply tensorflow_lite_support/python/task/audio/pybinds/_pywrap_audio_embedder.cc]: @khanhlvg Fixed it.

File: tensorflow_lite_support/python/task/audio/pybinds/BUILD (hunk 2 above)
Comment (kinaryml) [reply tensorflow_lite_support/python/task/audio/pybinds/BUILD]: Fixed.

File: tensorflow_lite_support/python/task/audio/core/pybinds/_pywrap_audio_buffer.cc [hunk 3]
```diff
@@ -40,15 +39,16 @@ PYBIND11_MODULE(_pywrap_audio_buffer, m) {
...
+            py::buffer buffer, const int sample_count,
+            const AudioBuffer::AudioFormat& audio_format)
+            -> std::unique_ptr<AudioBuffer> {
+              py::buffer_info info = buffer.request();
 
-        // TODO(b/220931229): Change this initializer to use AudioBuffer::Create
-        // and raise RuntimeError if initialization failed.
-        return absl::make_unique<AudioBuffer>(static_cast<float*>(info.ptr),
-                                              sample_count, audio_format);
-      }))
```
Comment (kinaryml) [inline tensorflow_lite_support/python/task/audio/core/pybinds/_pywrap_audio_buffer.cc:51]: @khanhlvg I've updated the `AudioBuffer` pybinds.

**Review by khanhlvg — APPROVED** [review]

File: tensorflow_lite_support/python/task/vision/pybinds/_pywrap_image_classifier.cc [hunk 4]
```diff
@@ -66,43 +63,42 @@ PYBIND11_MODULE(_pywrap_image_classifier, m) {
...
+                 .SerializeAsString());
              return classification_result;
            })
       .def("classify",
            [](ImageClassifier& self, const ImageData& image_data,
               const processor::BoundingBox& bounding_box)
-               -> tflite::support::StatusOr<processor::ClassificationResult> {
+               -> processor::ClassificationResult {
              // Convert from processor::BoundingBox to vision::BoundingBox as
-             // the latter is used in the C++ layer.
```
Comment (khanhlvg) [inline tensorflow_lite_support/python/task/vision/pybinds/_pywrap_image_classifier.cc:91]: FYI there are some typo fix reverted here, as well as above. I will fix them when importing the PR for review but please be more careful next time and pull the latest code from master. The code review happens internally using Google's code management system so sometimes I addressed internal feedback directly, resulting in some difference between the code in the pull request and what is merged into master.

Comment (kinaryml) [reply tensorflow_lite_support/python/task/vision/pybinds/_pywrap_image_classifier.cc:91]: @khanhlvg Alright, sorry about that. I'll keep that in mind.

**Pull request merged**